treeherder/model/models.py

```diff
@@ -706,6 +704,8 @@ class Meta:
             # this is intended to speed up queries for specific platform /
             # option collections on a push
             ('machine_platform', 'option_collection_hash', 'push'),
+            # speed up cycle data
+            ('repository', 'submit_time'),
```

Comment:
I think it actually makes more sense to cycle per-repository, as that lets us set a different expiry on them (i.e. in the future we might want to keep m-c data around longer than try).